app/components/listing_card_component.html.erb

```diff
@@ -50,6 +50,19 @@
           </svg>
         <% end %>
       <% end %>
+    <% when :transaction %>
+      <% if @transaction %>
+        <% owner = @current_user && @current_user.id == @listing.user_id %>
+        <div class="text-xs font-medium <%= @transaction.status == 'completed' ? 'text-green-600' : @transaction.status == 'in_progress' ? 'text-amber-600' : @transaction.status == 'cancelled' ? 'text-red-600' : 'text-gray-500' %>">
```

Comment:
This nested ternary is brutal. I think this would be worth moving into the component's rb file as a method.

```ruby
def transaction_status_color
  case @transaction.status
  when :completed then "text-green-600"
  ...
end
```

Then in your view you can just call the method to get the text color.